packages/vite/src/node/server/moduleGraph.ts

```diff
@@ -72,6 +72,13 @@ export class ModuleGraph {
     rawUrl: string,
     ssr?: boolean,
   ): Promise<ModuleNode | undefined> {
+    // Quick path, if we already have a module for this rawUrl (even without extension)
+    const cleanedUrl = removeImportQuery(removeTimestampQuery(rawUrl))
+    const mod = this.urlToModuleMap.get(cleanedUrl)
```

Comment:
Yes, that is a good idea. I also don't know if people are using `urlToModuleMap` for something in their plugins. There is a very scary error in previewjs https://github.com/vitejs/vite-ecosystem-ci/actions/runs/4552560932/jobs/8028053316#step:8:1977, that double discovery of a dependency doesn't seem related to this PR though 🤔 

I don't know about `rawUrlToModuleMap` as it is an optimization detail and I would prefer to avoid exposing it in the public API. I think I'll have this out of the class in a WeakMap against the module graph so it isn't accessible (if we use @internal, or _rawUrlToModuleMap, people may ignore us). What do you think?